bundles/org.openhab.binding.magentatv/src/main/java/org/openhab/binding/magentatv/internal/network/MagentaTVOAuth.java

```diff
@@ -0,0 +1,208 @@
+            httpResponse = HttpUtil.executeUrl(HTTP_POST, url, httpHeader, dataStream, null, NETWORK_TIMEOUT);
+            logger.trace("http response={0}", httpResponse);
+
+            OAuthTokenResponse token = gson.fromJson(httpResponse, OAuthTokenResponse.class);
+            if ((token.access_token == null) || token.access_token.isEmpty()) {
+                String errorMessage = MessageFormat.format("Authentication for account {0} failed: {1} (rc={2})",
+                        accountName, token.error_description, token.error);
+                throw new MagentaTVException(errorMessage);
+            }
+            accessToken = token.access_token;
+
+            // authenticateDevice
+            step = "authenticate with token";
+            logger.trace(step);
+            String uuid = UUID.randomUUID().toString();
+            String cnonce = MagentaTVControl.computeMD5(uuid);
+
+            // build url string
+            // {0} = uuid
+            // {1} = uuid
+            // {2} = uuid
+            // {3} = accessToken
+            // {4} = cnonce
+            // {5} = uuid
+            postData = MessageFormat.format(
+                    "'{'\"userType\": 1,\"terminalid\": \"{0}\",\"mac\":\"{1}\",\"terminaltype\":\"Iphone\",\"utcEnable\":1,\"timezone\":\"Europe/Berlin\",\"terminalvendor\":\"iPhone5\",\"osversion\":\"iOS10.3.3\",\"softwareVersion\":\"2.3.10.26\",\"terminalDetail\":['{'\"key\":\"HardwareSupplier\",\"value\":\"MyPhone\"'}','{'\"key\":\"DeviceClass\",\"value\":\"IPhone\"'}','{'\"key\":\"DeviceStorage\",\"value\":\"1\"'}','{'\"key\":\"DeviceStorageSize\",\"value\":12475'}','{'\"key\":\"GUID\",\"value\":\"{2}\"'}'],\"connectType\":1,\"reconnect\":true,\"accessToken\":\"{3}\",\"cnonce\":\"{4}\",\"caDeviceInfo\":['{'\"caDeviceType\":6,\"caDeviceId\":\"{5}\"'}'],\"preSharedKeyID\":\"NGTV000001\"'}'",
+                    // uuid, uuid, cnonce, uuid);
+                    "", "", "", accessToken, cnonce, "");
+            httpHeader.setProperty(HEADER_HOST, StringUtils.substringAfterLast(epghttpsurl, "/"));
+            httpHeader.setProperty(HEADER_CONTENT_TYPE, "text/plain;charset=UTF-8");
+            url = epghttpsurl + "/JSON/DTAuthenticate?SID=user&T=Iphone";
+            dataStream = new ByteArrayInputStream(postData.getBytes(Charset.forName("UTF-8")));
+            httpResponse = HttpUtil.executeUrl(HTTP_POST, url, httpHeader, dataStream, null, NETWORK_TIMEOUT);
+            logger.trace("http response={0}", httpResponse);
+            OAuthAutenhicateResponse resp = gson.fromJson(httpResponse, OAuthAutenhicateResponse.class);
+            if (resp.retcode.isEmpty() || !resp.retcode.equals("0")) {
+                retmsg = resp.desc;
+                String errorMessage = MessageFormat.format("Unable to authenticate: accountName={0}, rc={1} - {2}",
+                        accountName, retcode, retmsg);
+                logger.fatal(errorMessage);
```

Comment:
change to logger .debug

Reply:
this is very important, because it contains the error message of the OAuth login, e.g. wrong credentials, account banned etc.

Reply:
logger.warn then